ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -227,6 +247,23 @@ && getUserRateLimiterMap().get(user.getName()) != null
         }
     }
 
+    static Map<String, String> getParams(MLInput mlInput) throws IOException {
+        Map<String, String> result = new HashMap<>();
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        mlInput.getParameters().toXContent(builder, ToXContent.EMPTY_PARAMS);
+        builder.flush();
+        String json = builder.toString();
+
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Let's make a performance optimization here: 
1. Change the `MAPPER` here: `org.opensearch.ml.common.utils.StringUtils#MAPPER` to public.
2. Use the public MAPPER here for deserialization.